// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.gps_service";
option java_outer_classname = "GpsService";
option objc_class_prefix = "GPSS";

package gps_service;

service GpsService {
  rpc StreamLocation (StreamLocationRequest) returns (stream StreamLocationResponse) {}
}

message Point {
  int64 timestamp = 1;

  double latitude = 2;
  double longitude = 3;
  double altitude = 4;
  float speed = 5;
  float bearing = 6;

  float horizontal_accuracy_meters = 7;
  float vertical_accuracy_meters = 8;
  float speed_accuracy_mps = 9;
  float bearing_accuracy_degrees = 10;
}

message StreamLocationRequest {
}

message StreamLocationResponse {
  Point point = 1;
}
